plat/xen/arm/setup64.c

```diff
@@ -208,6 +208,52 @@ int uk_intctlr_plat_probe(void *arg)
 	return 0;
 }
 
+static int _get_ramdisk(struct ukplat_bootinfo *bi)
+{
+	int rc;
+	struct ukplat_memregion_desc mrd;
+	const __u64 *fdt_initrd_start;
+	const __u64 *fdt_initrd_end;
+	void *fdtp = HYPERVISOR_dtb;
+	int start_len, end_len;
+	__u64 initrd_base;
+	__u64 initrd_end;
+	int nchosen;
+
+	nchosen = fdt_path_offset(fdtp, "/chosen");
+	if (unlikely(nchosen < 0))
+		return 1;
+
+	fdt_initrd_start = fdt_getprop(fdtp, nchosen, "linux,initrd-start",
+				       &start_len);
+	if (unlikely(!fdt_initrd_start || start_len <= 0))
+		return 1;
+
+	fdt_initrd_end = fdt_getprop(fdtp, nchosen, "linux,initrd-end",
+				     &end_len);
+	if (unlikely(!fdt_initrd_end || end_len <= 0))
+		return 1;
+
+	initrd_base = fdt64_to_cpu(fdt_initrd_start[0]);
+	initrd_end = fdt64_to_cpu(fdt_initrd_end[0]);
+	mrd = (struct ukplat_memregion_desc){0};
```

Comment:
Maybe it can be done on the definition stage?
Smth like: struct ukplat_memregion_desc mrd = {0};